src/main/java/org/repodriller/scm/GitRepository.java

```diff
@@ -375,7 +382,7 @@ private synchronized void deleteMMBranch(Git git) throws GitAPIException {
 		}
 	}
 
-	public List<RepositoryFile> files() {
+	public synchronized List<RepositoryFile> files() {
```

Comment:
why did you synchronise it? I saw the comment in the previous PR, but now, I got curious!

Reply:
It can race with checkout/reset/etc.: reads on the FS state while the state is being mutated by those write operations.